### PR TITLE
docs(typehead): add demo for typeaheadNoResults output #3690

### LINKED ISSUE #3690: docs(typehead): add demo for typeaheadNoResults output
Add demo for typeaheadNoResults output (related #3342)

### PR DESCRIPTION
**Pull request by IraErshova**
Add demo for typeaheadNoResults output. To see message "No Results Found" enter the value of which is not in the list

Close #3690

# PR Checklist
Before creating new PR, please take a look at checklist below to make sure that you've done everything that needs to be done before we can merge it.

 - [x] read and followed the [CONTRIBUTING.md](https://github.com/valor-software/ngx-bootstrap/blob/development/CONTRIBUTING.md) guide.
 - [x] built and tested the changes locally.
 - [x] added/updated API documentation.
 - [x] added/updated demos.

### CODE REVIEW

File: demo/src/app/components/+typeahead/typeahead-section.list.ts [hunk 1]
```diff
@@ -77,6 +78,17 @@ export const demoComponentContent: ContentSection[] = [
         html: require('!!raw-loader?lang=markup!./demos/dropup/dropup.html'),
         outlet: DemoTypeaheadDropupComponent
       },
+      {
+        title: 'No result',
+        anchor: 'no-result',
+        description: `
+         <p>Fired on every key event and returns true in case of matches are not detected. To see message
+         "No Results Found" enter the value of which is not in the list</p>
```
Comment (IlyaSurmay) [inline demo/src/app/components/+typeahead/typeahead-section.list.ts]: enter the value of which is not in the list - > enter the value that doesn't match anything from the list

**Review by IlyaSurmay — CHANGES_REQUESTED** [review]

File: demo/src/app/components/+typeahead/demos/async/async.html [hunk 2]
```diff
@@ -1,12 +1,11 @@
...
+
+<input [(ngModel)]="asyncSelected"
+       [typeahead]="dataSource"
+       (typeaheadLoading)="changeTypeaheadLoading($event)"
+       (typeaheadOnSelect)="typeaheadOnSelect($event)"
+       [typeaheadOptionsLimit]="7"
+       typeaheadOptionField="name"
+       placeholder="Locations loaded with timeout"
+       class="form-control">
+<div *ngIf="typeaheadLoading===true">Loading</div>
```
Comment (IlyaSurmay) [inline demo/src/app/components/+typeahead/demos/async/async.html]: `*ngIf="typeaheadLoading"` ?

**Review by IlyaSurmay — CHANGES_REQUESTED** [review]

**Review by IlyaSurmay — DISMISSED** [review]

**Comment by YevheniiaMazur** [comment]
Tested, looks good

File: demo/src/app/components/+typeahead/typeahead-section.list.ts [hunk 3]
```diff
@@ -77,6 +78,17 @@ export const demoComponentContent: ContentSection[] = [
         html: require('!!raw-loader?lang=markup!./demos/dropup/dropup.html'),
         outlet: DemoTypeaheadDropupComponent
       },
+      {
+        title: 'No result',
+        anchor: 'no-result',
+        description: `
+         <p>Fired on every key event and returns true in case of matches are not detected. To see message
```
Comment (valorkin) [inline demo/src/app/components/+typeahead/typeahead-section.list.ts]: seriously? you are writing this for humans

**Review by valorkin — CHANGES_REQUESTED** [review]

**Review by valorkin — APPROVED** [review]

**Pull request merged**